pkgs/build-support/docker/default.nix

```diff
@@ -82,7 +82,7 @@ rec {
     export PATH=${shadow}/bin:$PATH
     mkdir -p /etc/pam.d
     if [[ ! -f /etc/passwd ]]; then
-      echo "root:x:0:0::/root:/bin/sh" > /etc/passwd
+      echo "root:x:0:0::/root:${stdenv.shell}" > /etc/passwd
```

Comment:
`/etc/shells` is only relevant for a user doing `chsh` on itself; `nologin` is usually not in `/etc/shells` but works as a login-no-intended shell just fine.